### ISSUE TITLE
rt1 - deep match in case fails with unpossible: tryCase

### ISSUE THREAD

**Issue by francisdb** [unisonweb/unison#352]
It seems to depend on the order of the cases and the use of this deep `Value.Bool false/true` match

This works as expected

```haskell
type Value = String Text
           | Bool Boolean

format : Value -> Text
format v = case v of
  Value.String s -> "\"" ++ s ++ "\""
  Value.Bool true -> "true"
  Value.Bool false -> "false"

> format (Value.String "foo")
> format (Value.Bool true)
```

output

```
  ┌
  │  10 | > format (Value.String "foo")
  │         ⧩
  │         "\"foo\""
  └
  ┌
  │  11 | > format (Value.Bool true)
  │         ⧩
  │         "true"
  └
```

but if we move the String case below the Bool ones the runtime crashes

```haskell
type Value = String Text
           | Bool Boolean

format : Value -> Text
format v = case v of
  Value.Bool true -> "true"
  Value.Bool false -> "false"
  Value.String s -> "\"" ++ s ++ "\""

> format (Value.String "foo")
> format (Value.Bool true)
```

output

```
unison: unpossible: tryCase ("foo", PatternB True)
CallStack (from HasCallStack):
  error, called at src/Unison/Runtime/Rt1.hs:467:17 in unison-parser-typechecker-0.1-7JbTx0c9MNfJWiedEv5WiM:Unison.Runtime.Rt1
```

**Comment by aryairani** [comment]
[added](https://github.com/unisonweb/unison/blob/e73647025839baaa2ff32822d5b105a50a73daaa/unison-src/tests/pattern-typing-bug.u) test case
```haskell
type Value = String Text
           | Bool Boolean

f : Value -> Nat
f v = case v of
  Value.Bool true -> 3
  _ -> 4

> f (Value.String "foo")
```
Should return `4`, not crash.

**Comment by aryairani** [comment]
Fixed in c6b5259bf91dc57e9532b83a493d76b3e93c0d8b.
